### PR TITLE
Add pipeline-level nf-tests

### LINKED ISSUE #61: Implement nf-test
### Description of feature

I'm going to switch the github ci to pytest-workflow so that github can monitor the consistency of pipeline output instead of just checking that PRs don't make the pipeline fail

### PR DESCRIPTION
**Pull request by atrigila**
Closes #61 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Amazing work, thank you. But any chance we can do it without the sharding? I've been 'discussing' with one or two people about it, and I'm not a fan. It's essentially really hard to work out what's being tested because nothing is named informatively.

I don't think we have enough components here for us to exceed the test limits and make sharding necessary anyway.

**Review by nschcolnicov — APPROVED** [review]
Looks great, love the new nf-test implementation, so much simple. Regarding @pinin4fjords comment about sharding, I also think it is easier to understand the tests, but I'm curious to know if we use sharding to make tests run faster or to save costs. If it is the latter, isn't it better to have it, even if reading the tests is a little bit harder?

**Comment by pinin4fjords** [comment]
> Looks great, love the new nf-test implementation, so much simple. Regarding @pinin4fjords comment about sharding, I also think it is easier to understand the tests, but I'm curious to know if we use sharding to make tests run faster or to save costs. If it is the latter, isn't it better to have it, even if reading the tests is a little bit harder?

As I understand it the sharding was introduced to bypass GitHub's limits on the numbers of independent tests, not out of any consideration to costs. In any case, my time is more valuable than computer time, so I'd like to save myself squinting at inscrutable test names.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Could you update the CHANGELOG please, following the conventions there (e.g. for some reason we put newer entries at the top ;-) ).

Then you'll have my tick :-)

**Comment by atrigila** [comment]
@pinin4fjords done :) thank you!

**Review by pinin4fjords — APPROVED** [review]
Brilliant- thanks for setting this up

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- [[#343](https://github.com/nf-core/differentialabundance/pull/343)] - Add pipeline-level nf-tests ([@atrigila](https://github.com/atrigila), review by [@pinin4fjords](https://github.com/pinin4fjords)) and [@nschcolnicov](https://github.com/nschcolnicov)
```
Comment (pinin4fjords) [inline CHANGELOG.md]:
```suggestion
- [[#343](https://github.com/nf-core/differentialabundance/pull/343)] - Add pipeline-level nf-tests ([@atrigila](https://github.com/atrigila), review by [@pinin4fjords](https://github.com/pinin4fjords)) and [@nschcolnicov](https://github.com/nschcolnicov))
```
oops, just a bracket

**Pull request merged**